tests/post/steps/conftest.py

```diff
@@ -6,29 +6,45 @@
 
 # Given {{{
 
-@given(parsers.parse("pods with label '{label}' are '{state}'"))
-def check_pod_state(request, host, k8s_client, label, state):
+@given(parsers.parse("pods with label '{label}' are '{expected_status}'"))
+def check_pod_status(request, host, k8s_client, label, expected_status):
     ssh_config = request.config.getoption('--ssh-config')
 
     # Helper to use retry utils
-    def _wait_for_state():
+    def _wait_for_status():
         pods = kube_utils.get_pods(
             k8s_client, ssh_config, label,
             namespace="kube-system", state="Running"
         )
         assert pods
+
+        for pod in pods:
+            # If really not ready, status may not have been pushed yet.
+            if pod.status.conditions is None:
+                assert expected_status == 'NotReady'
+                continue
+
+            # See https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions
+            MAP_STATUS = {
```

Comment:
This is duplicated, maybe put it in a common place so that we can use it from several functions.

Reply:
True, I will